### ISSUE TITLE
Allow excluding fields from the default search fields

### ISSUE THREAD

**Issue by adcin** [keepassxreboot/keepassxc#11072]
## Summary
- In the notes-section, there should be a possibility to exclude lines from the search. Similar to commenting out in source code.
- There should also be an option in the settings to exclude the whole notes field from search

### 1. Exclude from search by using special syntax

In this screenshot I use `//` to exclude a line, but it can be any prefix - like `!>` for example.

![keepass-unsearch](https://github.com/user-attachments/assets/1884d517-1701-4e69-a08c-ec6f725bedd5)

Now searching for "github" should NOT show this "Example" entry in the results.

As alternative a feature, to exclude enclosed text (e.g. `<!-you will never find this-!>`), no matter where it is located in an entry.

### 2. Option to exclude the notes field by default

![Screenshot_20240716_190454](https://github.com/user-attachments/assets/44e446c1-5dd4-419e-9094-1c01b0fcb77f)

## Context
In the notes I often write down which account I used for authentication, the associated email address or ssh key, code snippets and other stuff. And already when writing the note, I know that some of the stuff I never want to appear in search results. For example if I had this in the notes:
```
Repo: https://github.com/keepassxreboot/keepassxc.git
```
Then I would never want this entry to pop up ... especially not, when I'm searching for github.

And to be honest, I never want to appear results based by a match from the notes.

**Comment by droidmonkey** [comment]
Commenting out notes from search won't happen, we don't treat any data as special when searching as a principle. However, excluding notes from search would be useful.

**Comment by michaelk83** [comment]
There is already a search modifier to exclude a search term. So perhaps a new modifier to exclude a field?
Maybe the same exclude modifier but with just the field name without a string? So:
- `-n:github` => notes must not contain "github".
- `-n` => ignore notes.

**Comment by droidmonkey** [comment]
Yah that's a great idea

**Comment by adcin** [comment]
> There is already a search modifier to exclude a search term. So perhaps a new modifier to exclude a field? Maybe the same exclude modifier but with just the field name without a string? So:
> 
> * `-n:github` => notes must not contain "github".
> * `-n` => ignore notes.

That's a good idea, but I probably would still forget to use it and get annoyed. So how about additionally an option in the settings to define a search preset:

![Screenshot_20240716_205519](https://github.com/user-attachments/assets/9088ab06-6c61-4f87-b213-364e764e7464)

The search bar would then always be pre-filled with your favourite search preset (e.g. `title:`) and you still can delete it to get the normal blank field.

This could be also useful, if you are working on a specific project with a bunch of test accounts. Then you preset the search to something like "`n:lamda-testaccounts u:tester-no`" and have then only to type in the number of the username (`tester-no1`, `tester-no2`, ....) to get exactly the result you're looking for.

**Comment by michaelk83** [comment]
You can save specific searches, and access them from the bottom-left panel. So you can save whatever preset searches you want and load them from there.

**Comment by droidmonkey** [comment]
Agreed, search presets is already a feature and more robust and accessible through saved searches

**Comment by adcin** [comment]
> You can save specific searches, and access them from the bottom-left panel. So you can save whatever preset searches you want and load them from there.

I'm using KeePassXC since ages and have never seen this feature - I love it! Now I'm fan of your suggestion:

-n:github => notes must not contain "github".
-n => ignore notes.

**Comment by droidmonkey** [comment]
Not going to introduce this feature, use the existing search syntax. The more elegant solution is to actually search the fields you want to search, so for this example you would actually search for `t:github` to only search in the title field.

**Comment by michaelk83** [comment]
@droidmonkey we mostly agreed here to extend the search syntax in line with what already exists: https://github.com/keepassxreboot/keepassxc/issues/11072#issuecomment-2231588653
Is that not desirable anymore?

**Comment by droidmonkey** [comment]
The suggestion was not valid after further thought, the problem here is that there are certain "default fields" searched when no field specifier is given. There are other requests on here to expand the default fields to _everything_. If you want to narrow your search, the correct method is to pick the field you want to search and use it's prefix rather than excluding default fields.

**Comment by michaelk83** [comment]
It would make *more* sense with search everything by default, IMO, since one may want to search everything *except* a particular field. My suggestion also included syntax to exclude a term from a particular field, which isn't easy to do if you still want to search everything else.

Searching by a specific field has its own merit, but that's a separate use case.

**Comment by droidmonkey** [comment]
the syntax to exclude a term from a particular field is already implemented.

**Comment by michaelk83** [comment]
According to https://keepassxc.org/docs/KeePassXC_UserGuide#_search , you can exclude a term from all search fields, but not from a particular one. Which is technically the same if you search by only that field, but not the same if you search by multiple fields (or all by default).

**Comment by droidmonkey** [comment]
Not sure what you are talking about, this is _easily tested_ so recommend trying it out:

![Image](https://github.com/user-attachments/assets/6ee0891e-8632-4dcf-a8ac-5a87e31ff8a6)

**Comment by michaelk83** [comment]
Ah, ok. Didn't see that part. It's been a while since I read it in detail.

**Comment by adcin** [comment]
Feature already exists as search field syntax:
https://keepassxc.org/docs/KeePassXC_UserGuide#_search
